test/test_readme_sample.py

```diff
@@ -18,95 +18,69 @@
-        problem = ElectronicStructureProblem(driver)
-
-        # generate the second-quantized operators
-        second_q_ops = problem.second_q_ops()
-        main_op = second_q_ops[0]
-
-        particle_number = problem.grouped_property_transformed.get_property("ParticleNumber")
-
-        num_particles = (particle_number.num_alpha, particle_number.num_beta)
-        num_spin_orbitals = particle_number.num_spin_orbitals
-
-        # setup the classical optimizer for VQE
-        from qiskit.algorithms.optimizers import L_BFGS_B
-
-        optimizer = L_BFGS_B()
-
-        # setup the mapper and qubit converter
-        from qiskit_nature.mappers.second_quantization import ParityMapper
-        from qiskit_nature.converters.second_quantization import QubitConverter
-
-        mapper = ParityMapper()
-        converter = QubitConverter(mapper=mapper, two_qubit_reduction=True)
-
-        # map to qubit operators
-        qubit_op = converter.convert(main_op, num_particles=num_particles)
-
-        # setup the initial state for the ansatz
-        from qiskit_nature.circuit.library import HartreeFock
-
-        init_state = HartreeFock(num_spin_orbitals, num_particles, converter)
-
-        # setup the ansatz for VQE
-        from qiskit.circuit.library import TwoLocal
-
-        ansatz = TwoLocal(num_spin_orbitals, ["ry", "rz"], "cz")
-
-        # add the initial state
-        ansatz.compose(init_state, front=True, inplace=True)
-
-        # set the backend for the quantum computation
-        from qiskit import Aer
-
-        backend = Aer.get_backend("aer_simulator_statevector")
-
-        # setup and run VQE
-        from qiskit.algorithms import VQE
-
-        algorithm = VQE(ansatz, optimizer=optimizer, quantum_instance=backend)
-
-        result = algorithm.compute_minimum_eigenvalue(qubit_op)
-        print(result.eigenvalue.real)
-
-        electronic_structure_result = problem.interpret(result)
-        print(electronic_structure_result)
-
-        # ----------------------------------------------------------------------
-
-        self.assertAlmostEqual(result.eigenvalue.real, -1.8572750301938803, places=6)
+        # pylint: disable=exec-used
+
+        readme_name = "README.md"
+        readme_path = Path(__file__).parent.parent.joinpath(readme_name)
+        if not readme_path.exists() or not readme_path.is_file():
+            self.fail(msg=f"{readme_name} not found at {readme_path}")
+            return
+
+        readme_sample = None
+        with open(readme_path, encoding="UTF-8") as readme_file:
+            match_sample = re.search(
+                "```python.*```",
+                readme_file.read(),
+                flags=re.S,
+            )
+            if match_sample:
+                readme_sample = match_sample.group(0)[9:-3]
+
+        if readme_sample is None:
+            self.skipTest(f"No sample found inside {readme_name}.")
+            return
+
+        writer = Writer()
+        sys.stdout = writer
```

Comment:
I've started using [contextlib.redirect_stdout](https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stdout) for this purpose. See for example https://github.com/Qiskit/qiskit-nature/blob/main/test/results/test_electronic_structure_result.py#L30-L31

This will remove the need for the `Writer` class :+1: